examples/coalescing-bulkloader/src/main/java/com/github/benmanes/caffeine/examples/coalescing/bulkloader/CoalescingBulkloader.java

```diff
@@ -48,18 +49,22 @@
  */
 public class CoalescingBulkloader<Key, Value> implements AsyncCacheLoader<Key, Value> {
     private final Consumer<Collection<WaitingKey>> bulkLoader;
-    private int maxLoadSize; // maximum number of keys to load in one call
-    private long maxDelay; // maximum time between request of a value and loading it
-    private volatile Queue<WaitingKey> waitingKeys = new ConcurrentLinkedQueue<>();
-    private ScheduledExecutorService timer = Executors.newSingleThreadScheduledExecutor();
+    private final int maxLoadSize; // maximum number of keys to load in one call
+    private final long maxDelay; // maximum time between request of a value and loading it
+    private final Queue<WaitingKey> waitingKeys = new ConcurrentLinkedQueue<>();
+    private final ScheduledExecutorService timer = Executors.newSingleThreadScheduledExecutor();
     private ScheduledFuture<?> schedule;
     // Queue.size() is expensive, so here we keep track of the queue size separately
-    private AtomicInteger size = new AtomicInteger(0);
+    private final AtomicInteger size = new AtomicInteger(0);
 
-    private final class WaitingKey {
-        Key key;
-        CompletableFuture<Value> future;
-        long waitingSince;
+    class WaitingKey {
```

Comment:
I think this can be a static final class? It doesn't seem to need to retain the hidden reference to the parent.

Reply:
Tried that. Then it doesn't get the Key type parameter.
Let me try giving it it's own parameter.

Reply:
oh yes, you would need to define as `WaitingKey<K>` and use the diamond operator when newing it

Reply:
Yes, with it's own type parameters it can be static

Reply:
nice. ready for merge?